clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -320,6 +321,18 @@ static void canonicalizeIntrinsicThreeWayCmp(CIRBaseBuilderTy &builder,
   op.erase();
 }
 
+void LoweringPreparePass::lowerVAArgOp(VAArgOp op) {
+  CIRBaseBuilderTy builder(getContext());
+  builder.setInsertionPoint(op);
+
+  auto res = cxxABI->lowerVAArg(builder, op);
+  if (res) {
```

Comment:
I understand `lowerVAArg` could never return an empty value so the branching here is unnecessary.

Reply:
lowerVAArg returns null from LoweringPrepareItaniumCXXABI. This is the current way of prevent lowering of cir.var_arg if Lowering is not ready for an ABI (e.g. x86_64)